src/edu/mills/cs180a/StringUtilitiesTester.java

```diff
@@ -1,25 +1,53 @@
 package edu.mills.cs180a;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
 
 class StringUtilitiesTester {
-  @Test
-  void isSubstring_False_null() {
-    assertThrows(NullPointerException.class, () -> StringUtilities.isSubstring(null, "foo"));
-    assertThrows(NullPointerException.class, () -> StringUtilities.isSubstring("foo", null));
-    assertThrows(NullPointerException.class, () -> StringUtilities.isSubstring(null, null));
-  }
-
-  @Test
-  void isSubstring_False_EmptyString() {
-    assertTrue(StringUtilities.isSubstring("", "foo"));
-    // TODO: write rest
-  }
-
-  @Test
-  void isSubstring_True_Length1Substring() {
-    assertTrue(StringUtilities.isSubstring("A", "ABC"));
-  }
+    @Test
+    void isSubstring_False_null() {
+        assertThrows(NullPointerException.class, () -> StringUtilities.isSubstring(null, "foo"));
+        assertThrows(NullPointerException.class, () -> StringUtilities.isSubstring("foo", null));
+        assertThrows(NullPointerException.class, () -> StringUtilities.isSubstring(null, null));
+    }
+
+    @Test
+    void isSubstring_False_EmptyString() {
+        assertTrue(StringUtilities.isSubstring("", "foo"));
+        assertTrue(StringUtilities.isSubstring("", "bar"));
+        assertTrue(StringUtilities.isSubstring("", ""));
+    }
+
+    @Test
+    void isSubstring_True_Length1Substring() {
+        assertTrue(StringUtilities.isSubstring("A", "ABC"));
+    }
+
+    @Test
+    void isSubstring_False_Length1Substring() {
+        assertFalse(StringUtilities.isSubstring("A", "DEF"));
+    }
+
+    @Test
+    void getLength_True_Length6_4_9String() {
```

Comment:
Your other names are very good, but this isn't the right format. First, `getLength()` doesn't return `true`. Second, `Length6_4_9` isn't a good description. Why not `NonZeroLengths`?

Reply:
Great, I changed the name.